lgc/builder/DescBuilder.cpp

```diff
@@ -74,6 +74,11 @@ Value *DescBuilder::CreateLoadBufferDesc(unsigned descSet, unsigned binding, Val
       return UndefValue::get(getBufferDescTy(pointeeTy));
     }
 
+    if (node->binding != binding) {
+      // We found a binding in the middle of a range. Add the difference to the index.
+      descIndex = CreateAdd(descIndex, getInt32(binding - node->binding));
```

Comment:
I don't know LLVM well enough, but will descIndex be a constant if the two inputs to CreateAdd are constants?  If not then the next if condition will always be false.  You should have a test that tests both the descIndex being constant and a variable.

Reply:
Line 99 will be executed by other tests, but `binding - node->binding` will always be 0 in those cases.  We need a test where it is not 0.

Reply:
> It looks like this branch is never hit in the tests

Right, if the descriptor being loaded is not a buffer descriptor, this function will not be called.  The test has a range of Resource descriptors.  To test the code in this function, you need a range of buffer descriptors.